lib/improver/nbhood/recursive_filter.py

```diff
@@ -73,29 +79,35 @@ def __init__(self, alpha_x=None, alpha_y=None, iterations=None,
                 originally masked.
 
         Raises:
-            ValueError: If alpha_x is not set such that 0 < alpha_x < 1
-            ValueError: If alpha_y is not set such that 0 < alpha_y < 1
+            ValueError: If alpha_x is not set such that 0 < alpha_x <= 0.5
+            ValueError: If alpha_y is not set such that 0 < alpha_y <= 0.5
             ValueError: If number of iterations is not None and is set such
-                        that iterations is not >= 1
+                        that iterations is less than 1.
 
-        """
-        if alpha_x is not None:
-            if not 0 < alpha_x < 1:
-                raise ValueError(
-                    "Invalid alpha_x: must be > 0 and < 1: {}".format(
-                        alpha_x))
+        Warns:
+            UserWarning:
+                If iterations is higher than 2.
 
-        if alpha_y is not None:
-            if not 0 < alpha_y < 1:
-                raise ValueError(
-                    "Invalid alpha_y: must be > 0 and < 1: {}".format(
-                        alpha_y))
 
+        """
+        alpha_error = ("alpha must be less than 0.5. A large alpha value"
+                       "leads to poor conservation of probabilities: ")
+        if alpha_x is not None and not 0 < alpha_x <= 0.5:
```

Comment:
(We could ~halve the number of lines of code here by doing this in a loop)

Reply:
Looped it up.

Not sure about moving it all to process, yes it is being split, but that is where the arguments are being given to the object. There is no consistent way of this being done throughout the code.

Reply:
Both `__init__` and `process` could call a function containing this bit?